src/braket/circuits/gates.py

```diff
@@ -999,7 +999,7 @@ def to_ir(self, target: QubitSet):
         return ir.CZ.construct(control=target[0], target=target[1])
 
     def to_matrix(self) -> np.ndarray:
-        return np.diag([1.0, 1.0, 1.0, -1.0])
+        return np.diag([complex(1.0), 1.0, 1.0, -1.0])
```

Comment:
I had to add this because the resulting matrix was not actually composed of `complex` numbers.